runhouse/servers/cluster_servlet.py

```diff
@@ -27,6 +29,34 @@ def __init__(
         self._key_to_env_servlet_name: Dict[Any, str] = {}
         self._auth_cache: AuthCache = AuthCache()
 
+        if self.cluster_config.get("resource_subtype", None) == "OnDemandCluster":
+            self._last_activity = time.time()
+            self._last_register = None
+            thread = threading.Thread(target=self.update_autostop, daemon=True)
+            thread.start()
+
+    ##############################################
+    # Cluster autostop
+    ##############################################
+    def update_autostop(self):
+        from sky.skylet import configs as sky_configs
+
+        while True:
+            autostop = self.cluster_config.get("autostop_mins", -1)
+            self._last_register = sky_configs.get_config("autostop_last_active_time")
+            if autostop > 0 and (
+                not self._last_register
+                or (
+                    # within 5 min of autostop and there's more recent activity
+                    autostop - (time.time() - self._last_register) / 60 < 5
```

Comment:
@dongreenberg now that its running in the background do you think it's still useful to check within x min of autostopping or just updating every time last_activity > last_register

Reply:
I think what you have is fine, we can probably make it just 2 minutes though, not 5.